src/Scripting/Core/Hosting/ObjectFormatter/CommonTypeNameFormatter.cs

```diff
@@ -93,7 +93,7 @@ public virtual string FormatTypeArguments(Type[] typeArguments, CommonTypeNameFo
 
             if (typeArguments.Length == 0)
             {
-                return "";
+                throw new ArgumentException(nameof(typeArguments));
```

Comment:
@tmat, converting all the other argument validation to asserts is out of scope for this change, so I'm just going to leave the exception for now.